src/QirTools/pyqir/qirlib/src/interop.rs

```diff
@@ -0,0 +1,121 @@
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct QuantumRegister {
+    pub name: String,
+    pub index: u64,
+}
+
+impl QuantumRegister {
+    pub fn new(name: String, index: u64) -> Self {
+        QuantumRegister { name, index }
+    }
+
+    pub fn as_register(&self) -> Register {
+        Register::Quantum(self.clone())
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ClassicalRegister {
+    pub name: String,
+    pub size: u64,
+}
+
+impl ClassicalRegister {
+    pub fn new(name: String, size: u64) -> Self {
+        ClassicalRegister { name, size }
+    }
+
+    pub fn as_register(&self) -> Register {
+        Register::Classical(self.clone())
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Register {
+    Quantum(QuantumRegister),
+    Classical(ClassicalRegister),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Controlled {
+    pub control: String,
+    pub target: String,
+}
+
+impl Controlled {
+    pub fn new(control: String, target: String) -> Self {
+        Controlled { control, target }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Rotated {
+    pub theta: f64,
+    pub qubit: String,
+}
+
+impl Rotated {
+    pub fn new(theta: f64, qubit: String) -> Self {
+        Rotated { theta, qubit }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Single {
+    pub qubit: String,
+}
+
+impl Single {
+    pub fn new(qubit: String) -> Self {
+        Single { qubit }
+    }
+}
+
+// https://github.com/microsoft/qsharp-language/blob/ageller/profile/Specifications/QIR/Base-Profile.md
+#[derive(Clone, Debug, PartialEq)]
+pub enum Instruction {
+    Cx(Controlled),
+    Cz(Controlled),
+    H(Single),
+    M { qubit: String, target: String },
+    Reset(Single),
+    Rx(Rotated),
+    Ry(Rotated),
+    Rz(Rotated),
+    S(Single),
+    Sdg(Single),
```

Comment:
It looks like `Sdg` and `Tdg` aren't used in Base-Profile.md, but rather `s_adj` and `t_adj` such that it should be possible to use `SAdj` and `TAdj` here while still being consistent with that spec.